docs/actions.md

```diff
@@ -1,182 +1,129 @@
+      />
+    </div>,
   actions: {
-    up: ({ count, maxCount }) => ({
-      count: count + (maxCount > count ? 1 : -maxCount)
-    })
+    setText: (state, actions, text) => ({ text })
   }
 })
 ```
 
-You can pass data to actions as well.
-
-```jsx
-app({
-  state: {
-    count: 0
-  },
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state.count}
-      </h1>
-      <button onclick={() => actions.up(1)}>+</button>
-    </main>,
-  actions: {
-    up: ({ count }, actions, data = 0) => ({
-      count: count + data
-    })
-  }
-})
-```
+Actions are often called as a result of user events triggered on the [view](/docs/view.md) or inside application [events](/docs/events.md).
 
-## Side Effects
+## Thunks
 
-Actions are not required to have a return value. You can use them to call other actions, for example after an async operation has completed.
+Actions can return a function instead of a partial state. This function is called a _thunk_. They operate like regular actions but will not trigger a state update unless you call [`update`](/docs/api.md#update) from within the thunk function.
 
 ```jsx
 app({
-  state: {
-    count: 0
-  },
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state.count}
-      </h1>
-      <button onclick={actions.upLater}>+</button>
-    </main>,
   actions: {
-    up: ({ count }) => ({
-      count: count + 1
-    }),
-    upLater: (state, actions) => {
-      setTimeout(actions.up, 1000)
+    defer(state, actions, data) {
+      return update => {
+        // ...
+        update(newData)
+      }
     }
   }
 })
 ```
 
-Actions can return a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+The action returns the result of the thunk, allowing you to modify how actions operate and what types they can return.
+
+Use thunks to defer state updates, create [getters](#getters), scoped mixins, etc.
+
+## Async Updates
+
+Use [thunks](#thunks) to update the state asynchronously, e.g., after a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) resolves.
+
+[Try it Online](https://codepen.io/hyperapp/pen/ZeByKv?editors=0010)
 
 ```jsx
 app({
-  state: 0,
-  view: (state, actions) =>
-    <main>
-      <h1>
-        {state}
-      </h1>
-      <button onclick={actions.upLater}>+1</button>
-    </main>,
   actions: {
-    upLater: (state, actions) =>
-      new Promise(resolve => setTimeout(resolve, 1000, state + 1))
+    getURL(state) {
+      return update => fetch(`/search?q=${state.query}`)
+        .then(data => data.json())
+        .then(json => update({
+          url: json[0].url
+        })
+      )
+    }
   }
 })
 ```
 
-Actions can be written as [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) too.
+Actions need not have a return value at all. You can use them to create side effects, call other actions, etc.
```

Comment:
Perhaps worth mentioning explicitly that..
`Actions that return nothing will not trigger a state update`